Assets/EditorTests/LobbyTests.cs

```diff
@@ -0,0 +1,199 @@
+                {
+                    // The test will fail, as the GameObject SHOULD have had been on a specific layer, but WAS NOT.
+                    // We are delaying the assertion so that all errors are identified in the console, rather than requiring the
+                    // test to be run multiple times (and only identifying a single breach each time).
+                    allRelevantMonoBehavioursAreOnTheRightLayer = false;
+                    sb.Append($"-> {monoType.Name} script requires scene object '{mono.gameObject.name}' to be on {attribute.layer} layer, but it was on {LayerMask.LayerToName(mono.gameObject.layer)} layer.\n");
+                }
+            }
+        }
+
+        // ASSERT
+        Assert.IsTrue(allRelevantMonoBehavioursAreOnTheRightLayer, sb.ToString());
+    }
+
+    /// <summary>
+    /// Test to confirm that prefabs within the project are on the correct layers.
+    /// The purpose of this is to ensure that any prefabs instantiated at runtime are added to the correct layer.
+    /// </summary>
+    [Test]
+    public void SpecifiedMonoBehavioursWithinPrefabsAreOnTheirMandatedLayers()
+    {
+
+        // ARRANGE
+        bool allRelevantMonoBehavioursAreOnTheRightLayer = true;
+        StringBuilder sb = new StringBuilder();
+        MonoBehaviour[] prefabMonoBehaviours;
+
+        // ACT - Check each MonoBehaviour in the scene
+
+        foreach (GameObject prefab in allPrefabs)
+        {
+            prefabMonoBehaviours = prefab.GetComponentsInChildren<MonoBehaviour>();
+
+            foreach (MonoBehaviour mono in prefabMonoBehaviours)
+            {
+                Type monoType = mono.GetType();
+                RequiredLayerAttribute attribute = Attribute.GetCustomAttribute(monoType, typeof(RequiredLayerAttribute)) as RequiredLayerAttribute;
+                if (attribute != null)
+                {
+                    // Once we are here, we have found a MonoBehaviour with a RequiredLayerAttribute.
+                    // We now need to test the GameObject to see if it is on the layer that is mandated.
+
+                    if (mono.gameObject.layer != LayerMask.NameToLayer(attribute.layer))
+                    {
+                        // The test will fail, as the GameObject SHOULD have had been on a specific layer, but WAS NOT.
+                        // We are delaying the assertion so that all errors are identified in the console, rather than requiring the
+                        // test to be run multiple times (and only identifying a single breach each time).
+                        allRelevantMonoBehavioursAreOnTheRightLayer = false;
+                        sb.Append($"-> {monoType.Name} script requires object '{mono.gameObject.name}' (in prefab {prefab.name}) to be on {attribute.layer} layer, but it was on {LayerMask.LayerToName(mono.gameObject.layer)} layer.\n");
+                    }
+                }
+            }
```

Comment:
This code is duplicated from the test above. Can you extract it?

Reply:
Haha you caught me! I had the same thought when I wrote it, but didn't think anyone would notice / care. I've added a commit to tidy it up.